### ISSUE TITLE
Please migrate away from distutils

### ISSUE THREAD

**Issue by hosiet** [mypaint/mypaint#1228]
### Description of bug

Python 3.12+ has officially purged distutils from the std library. All distutils functionality can be provided by setuptools. Please consider completely remove distutils usage from the repo (most importantly in setup.py).

### Steps to reproduce

Build the package with python 3.12

### Backtraces or error messages

```shell
File "/<<PKGBUILDDIR>>/setup.py", line 17, in <module>
     from distutils.command.build import build
ModuleNotFoundError: No module named 'distutils'
E: pybuild pybuild:391: clean: plugin distutils failed with: exit
code=1: python3.12 setup.py clean
```

### MyPaint version

git HEAD

### Operating System(s)

N/A

### Desktop Environment (Linux)

N/A

### Code of Conduct

- [X] I acknowledge that my conduct contributes to shaping the community.

**Comment by hosiet** [comment]
Looking at code search from https://github.com/search?q=repo%3Amypaint%2Fmypaint%20distutils&type=code , there are usages in setup.py https://github.com/mypaint/mypaint/blob/f95972f23c6e8313042fff80ed1249454e357b0d/setup.py#L28-L29 and tests/unported/performance.py https://github.com/mypaint/mypaint/blob/f95972f23c6e8313042fff80ed1249454e357b0d/tests/unported/performance.py#L9 https://github.com/mypaint/mypaint/blob/f95972f23c6e8313042fff80ed1249454e357b0d/tests/unported/performance.py#L450

**Comment by AesaraB** [comment]
Realistically this will be something I get to in one or two weeks as I'm currently working on refactoring the documentation

**Comment by eeickmeyer** [comment]
@AesaraB First of all, thank you so much for this work! I know quite a few people that love mypaint and use it with their Wacom tablets on Ubuntu Studio!

Unfortunately, this has officially been removed from Ubuntu Noble Numbat (future 24.04 LTS) in order to accommodate the Python 3.12 transition. I do hate to rush this, but the sooner a fix can be provided, the better. I went to update the seeds and metapackages for Ubuntu Studio just now and, much to my surprise (and alarm!), I found it had been removed.

**Comment by AesaraB** [comment]
As a temporary holdover, see if the AppImage works

**Comment by AesaraB** [comment]
OH! I apologise, I hadn't realised you were a packager. There are a number of issues to contend with before the next minor release (there have been hundreds of commits since then, plus this project has more or less been abandoned for two years, so some spring cleaning too). I could, for your purposes, create a patch file that you can apply to relevant file for the current release.

**Comment by eeickmeyer** [comment]
Yes! Myself and @hosiet are both packages, me for Ubuntu, himself for Debian. :) All we need are patches and we're good to go! :smile:

**Comment by AesaraB** [comment]
[PEP 632 – Deprecate distutils module](https://peps.python.org/pep-0632/)

While Python 3.12 will not include the distutils library, [``setuptools`` will](https://github.com/pypa/setuptools/issues/2928#issuecomment-997138589).

- ``distutils.command.build.build`` is grandfathered into [setuptools](https://github.com/pypa/setuptools/blob/main/setuptools/command/build.py).
  - [Setuptools may eventually include its own implementation of custom build steps.](https://github.com/pypa/setuptools/issues/2591)
- ``distutils.command.clean.clean`` does not have a modern equivalent, as [setup.py is depreciated](#1229).
  - We'll follow the convention in the current master branch for this. 
  https://github.com/mypaint/mypaint/blob/f95972f23c6e8313042fff80ed1249454e357b0d/setup.py#L26-L29

This patch modifies two files as the bare minimum I need to get MyPaint built. It:
1. Makes appropriate fixes for the problems listed aboev
2. Fixes a depreciated (and removed) call in gettext.
3. **Does not** yet make changes to the unit tests, I'm hoping they'll just throw a warning at you, rather than stop you from building anything.

Patch file below:
[distutils.patch.txt](https://github.com/mypaint/mypaint/files/14106009/distutils.patch.txt)

**Comment by AesaraB** [comment]
FYI, I haven't tested this on Python 3.12 as my development environment doesn't package it in the default repos. I'll set things up properly if setup.py is still broken.

**Comment by hosiet** [comment]
Thanks for the patch.

@eeickmeyer please note, I cannot make any influence on the Ubuntu side. You may need more communication with Ubuntu internals so that package mypaint won't miss Ubuntu 24.04 release deadline.

**Comment by AesaraB** [comment]
Hi, this issue is a subset of #1229, and no longer needs to be open. Can you confirm/deny the patch as working?

**Comment by hosiet** [comment]
Ignoring the test scripts, I believe the patch works for package building. I think we can close the issue and focus more on #1229 for the next release. Thanks!
